pkg/client/containerjfr_client.go

```diff
@@ -194,3 +191,20 @@ func (client *ContainerJfrClient) syncMessage(msg *CommandMessage, responsePaylo
 	}
 	return err
 }
+
+func (client *ContainerJfrClient) readResponse(msg *CommandMessage,
+	responsePayload interface{}) (*ResponseMessage, error) {
+	// Set the original message ID so our decoder can verify it.
+	// By setting the output argument in the struct, the decoder knows what type
+	// to unmarshall the payload into and also stores it for returning to the caller.
+	resp := &ResponseMessage{ID: msg.ID, Payload: responsePayload}
+
+	// Keep reading messages until we find a response to this message,
+	// or encounter an error other than ErrWrongID
+	for {
+		err := client.conn.ReadJSON(resp)
+		if err == nil || err != ErrWrongID {
```

Comment:
Would it make sense to log the `ErrWrongID`s at some lower log level? It seems right now that if somehow the operator does end up stuck in this loop that it'll be difficult to diagnose it as being in that state.

I do agree that some form of timeout would also make sense here. Maybe 10 seconds? 30? Most commands should have very short response latencies, but it also depends on various external factors - since we're talking about an Operator here then there is most likely an authentication system enabled, so message response times are also subject to authz latencies, for example.

Reply:
I've changed this to a debug-level message, and also added an argument to the dev_operator setting the log level to debug.

I added a 30 second timeout for WebSocket reads and writes, and also used a 30 second timeout to exit the read loop. We could still get a 29 second read with a wrong ID and a 30 second one that times out, but the upper bound should be < 60 seconds. Still, this prevents it from hanging indefinitely.

Reply:
I should point out that the logging happens in UnmarshalJSON within `command_types.go`, right before `ErrWrongID` is returned.